accounts-db/src/append_vec.rs

```diff
@@ -765,19 +767,25 @@ pub mod tests {
     }
 
     #[test]
-    #[should_panic(expected = "assertion failed:")]
+    // rust 1.73+ (our as-of-writing nightly version) changed panic message. we're stuck with this
+    // short common substring until the monorepo is fully 1.73+ including stable.
+    #[should_panic(expected = "left == right")]
```

Comment:
hehe, `assertion failed:` is arguably out of rust's compatibility guarantee. ideally, custom user-supplied (i.e. controlled by us) string should be specified at proper `assert!` and `should_panic` should be adjusted as such. but i'm lazy, you know. ;)

before rust 1.73:

```
thread 'append_vec::tests::test_storable_accounts_with_hashes_and_write_versions_new3' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`', accounts-db/src/account_storage/meta.rs:56:9
```

after rust 1.73:
```
thread 'append_vec::tests::test_storable_accounts_with_hashes_and_write_versions_new3' panicked at accounts-db/src/account_storage/meta.rs:56:9:
assertion `left == right` failed
  left: 1
 right: 0
```

kudos to https://github.com/rust-lang/rust/pull/111071, i think this is cleaner indeed. :)

Reply:
> `assertion failed:` is arguably out of rust's compatibility guarantee

soooooo, i'm the bad guy, who started this thing, atoning my own sin now after 3.5 years..... lol:

```
$ git log -S "assertion failed: " --reverse -p | grep -C2 -E "(^commit|^Author)|assertion failed: " | head -n 12
commit 7614af2a457a1a0d142a657aae36287a53374811
Author: Ryo Onodera <ryoqun@gmail.com>
Date:   Tue Feb 11 16:46:33 2020 +0900

--
+
+    #[test]
+    #[should_panic(expected = "assertion failed: self.is_frozen()")]
+    fn test_verify_hash_unfrozen() {
+        let (genesis_config, _mint_keypair) = create_genesis_config(2_000);
--
     // Test that two bank forks with the same accounts should not hash to the same value.
```

bf735721f94f488e16474a5e0c3b1064e01844ef